marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/graph/searchNodes.sjs

```diff
@@ -128,60 +128,78 @@ let edges = [];
     nodeOrigin.id = item.subjectIRI;
-    nodeOrigin.label = item.subjectLabel;
+    nodeOrigin.label = subjectLabel;
     nodeOrigin.additionaProperties = null;
-    nodeOrigin.group = item.predicateIRI;
+    nodeOrigin.group = group;
     nodeOrigin.isConcept = false;
     nodeOrigin.count = 1;
     nodes[item.subjectIRI] = nodeOrigin;
   } else {
     nodeOrigin = nodes[item.subjectIRI];
-    nodeOrigin.label = item.subjectLabel || nodeOrigin.label;
-    nodeOrigin.group = item.predicateIRI || nodeOrigin.group;
+    nodeOrigin.label = subjectLabel || nodeOrigin.label;
+    nodeOrigin.group = group;
     nodeOrigin.additionaProperties = null;
     nodes[item.subjectIRI] = nodeOrigin;
   }
 
   if (item.nodeCount && item.nodeCount >= 1) {
-    let edge = {};
+    //Gather object node data as if count is 1
     let objectIRI = item.firstObjectIRI.toString();
+    let objectIRIArr = objectIRI.split("/");
+    if (item.firstObjectLabel === null) {
+      item.firstObjectLabel = objectIRIArr[objectIRIArr.length - 1];
+    }
     let objectLabel = item.firstObjectLabel.toString();
-    if (item.nodeCount > 1) { //If there are more than one we link the edge to the object entity type
-      let entityIRIArr = objectIRI.split("/");
-      let entityType = entityIRIArr[entityIRIArr.length - 2].split("-");
+    let objectId = item.firstObjectIRI.toString();
+    let objectGroup = objectIRI.substring(0, objectIRI.length - objectIRIArr[objectIRIArr.length - 1].length - 1);
+
+    //Override if count is more than 1. We will have a node with badge.
+    if (item.nodeCount > 1) {
+      let entityType = objectIRIArr[objectIRIArr.length - 2];
       objectIRI = entityType;
       objectLabel =  entityType;
+      objectId = item.subjectIRI.toString() + "-" + firstObjectArr[firstObjectArr.length - 2];
     }
+
+    let edge = {};
     edge.id = "edge-" + item.subjectIRI + "-" + item.predicateIRI + "-" + objectIRI;
 
     let predicateArr = item.predicateIRI.toString().split("/");
     let edgeLabel = predicateArr[predicateArr.length - 1];
     edge.label = edgeLabel;
-
     edge.from = item.subjectIRI;
-    edge.to = objectIRI;
+    edge.to = objectId;
     edges.push(edge);
 
-    if (!nodes[item.objectIRI]) {
+    if (!nodes[objectId]) {
+
       let objectNode = {};
-      objectNode.id = item.subjectIRI + "-" + objectIRI;
+      objectNode.id = item.firstObjectIRI;
       objectNode.label = objectLabel;
-      objectNode.group = null;
+      objectNode.group = objectGroup;
       objectNode.isConcept = false;
       objectNode.count = item.nodeCount;
-      nodes[item.firstObjectIRI] = objectNode;
+      nodes[objectId] = objectNode;
     }
   }
   else if (item.predicateIRI !== undefined && item.predicateIRI.toString().length > 0){
       let edge = {};
 
       let predicateArr = item.predicateIRI.toString().split("/");
       let edgeLabel = predicateArr[predicateArr.length - 1];
+      edge.id = "edge-" + item.subjectIRI + "-" + item.predicateIRI;
```

Comment:
Found a small issue with repeated ids on some edges. Could you push the following fix?
Full line should be:
` edge.id = "edge-" + item.subjectIRI + "-" + item.predicateIRI + "-" + item.objectIRI;`

Reply:
Done.